spec/models/post_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it 'is valid with a title, comments_counter, and likes_counter within valid ranges' do
+    user = User.create(name: 'Tom', post_counter: 0)
+    post = user.posts.new(title: 'My Post', comments_counter: 0, likes_counter: 0)
+    expect(post).to be_valid
+  end
+
+  it 'is not valid without a title' do
+    user = User.create(name: 'Tom', post_counter: 0)
+    post = user.posts.new(title: nil)
+    post.valid?
+    expect(post.errors[:title]).to include("can't be blank")
+  end
+
+  it 'is not valid with a title exceeding 250 characters' do
+    user = User.create(name: 'Tom', post_counter: 0)
+    post = user.posts.new(title: 'A' * 251)
+    post.valid?
+    expect(post.errors[:title]).to include('is too long (maximum is 250 characters)')
+  end
+
+  it 'is not valid with a negative comments_counter' do
+    user = User.create(name: 'Tom', post_counter: 0)
+    post = user.posts.new(title: 'My Post', comments_counter: -1, likes_counter: 0)
+    post.valid?
+    expect(post.errors[:comments_counter]).to include('must be greater than or equal to 0')
+  end
+
+  it 'is not valid with a negative likes_counter' do
+    user = User.create(name: 'Tom', post_counter: 0)
+    post = user.posts.new(title: 'My Post', comments_counter: 0, likes_counter: -1)
+    post.valid?
+    expect(post.errors[:likes_counter]).to include('must be greater than or equal to 0')
+  end
+
+  it 'updates the posts counter for a user' do
+    user = User.create(name: 'Tom', post_counter: 0)
+    post = user.posts.create(title: 'My Post', comments_counter: 0, likes_counter: 0)
+
+    post.increment_user_post_counter
+
+    user.reload
+    expect(user.post_counter).to eq(2)
+  end
+
```

Comment:
---

Hello,

You've done an excellent job testing the Post model and ensuring that it adheres to the validation requirements outlined for the project. Here's a breakdown of the review based on your code:

🛠️ **Recommendations:**
- It's not clear if you are testing the datatype of `comments_counter` and `likes_counter` to ensure they are integers. Consider adding tests to confirm that these fields reject non-integer values.

- In your model (`post.rb`), ensure you have the following validations to pass the tests:

```ruby
# app/models/post.rb
class Post < ApplicationRecord
  validates :title, presence: true, length: { maximum: 250 }
  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
end
```

Overall, you're demonstrating a solid understanding of RSpec and Rails model validations. Keep up the great work, and remember to align your tests with the exact behavior of your methods!

---